plugin-api/src/main/java/org/sonar/api/batch/sensor/rule/NewAdHocRule.java

```diff
@@ -59,6 +61,15 @@ public interface NewAdHocRule {
    */
   NewAdHocRule type(RuleType type);
 
+  /**
+   * Attribute of the rule according to Clean Code Taxonomy.
+   * Providing it is optional for now but will become mandatory in the future.
+   *
+   * @since 10.0
+   */
+  @CheckForNull
```

Comment:
What is the goal of `@CheckForNull` here

Reply:
During the transition of CTT, the cleanCodeAttribute is not mandatory. I mean, when we use this in our product, we should check if this field is provided or not before use it.
AdHocRule are mainly used for third-party library (If I'm right), I guess they will not provide this field for a little time.

Reply:
The function returns `NewAdHocRule`, not `cleanCodeAttribute`, so I doubt it can be null

Reply:
Good Catch. I confused with AdHocRule. fixed.